inspirehep/modules/theme/views.py

```diff
@@ -145,6 +149,53 @@ def data():
 def ping():
     return 'OK'
 
+#
+# Handlers for AJAX requests regarding references and citations
+#
+
+
+@blueprint.route('/ajax/references', methods=['GET'])
+def ajax_references():
+    """Handler for datatables references view"""
+
+    if request.args.get('recid'):
+        recid = request.args.get('recid', '')
+        collection = request.args.get('collection', '')
+
+    from flask import jsonify
+    from inspirehep.utils.references import Reference
+    from invenio_pidstore.models import PersistentIdentifier
+
+    pid = PersistentIdentifier.get(collection, recid)
```

Comment:
you could use this utility function https://github.com/inspirehep/inspire-next/blob/invenio3/inspirehep/utils/record_getter.py#L65